integrationtests/compatibility-mode-it/src/test/java/org/infinispan/it/compatibility/DistL1EmbeddedHotRodTest.java

```diff
@@ -0,0 +1,76 @@
+package org.infinispan.it.compatibility;
+
+import org.infinispan.Cache;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.distribution.MagicKey;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.TestingUtil;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.getSplitIntKeyForServer;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+@Test(groups = "functional", testName = "it.compatibility.DistL1EmbeddedHotRodTest")
+public class DistL1EmbeddedHotRodTest extends AbstractInfinispanTest {
+
+   private static final int NUM_OWNERS = 1;
+
+   private CompatibilityCacheFactory<Integer, String> cacheFactory1;
+   private CompatibilityCacheFactory<Integer, String> cacheFactory2;
+
+   @BeforeClass
+   protected void setup() throws Exception {
+      cacheFactory1 = new CompatibilityCacheFactory<Integer, String>(CacheMode.DIST_SYNC, NUM_OWNERS, true).setup();
+      cacheFactory2 = new CompatibilityCacheFactory<Integer, String>(CacheMode.DIST_SYNC, NUM_OWNERS, true)
+            .setup(cacheFactory1.getHotRodPort(), 100);
+
+      List<Cache<Integer, String>> caches = Arrays.asList(cacheFactory1.getEmbeddedCache(), cacheFactory2.getEmbeddedCache());
+      TestingUtil.blockUntilViewsReceived(30000, caches);
+      TestingUtil.waitForRehashToComplete(caches);
+
+      assertTrue(cacheFactory1.getHotRodCache().isEmpty());
+      assertTrue(cacheFactory2.getHotRodCache().isEmpty());
+   }
+
+   @AfterClass
+   protected void teardown() {
+      CompatibilityCacheFactory.killCacheFactories(cacheFactory1, cacheFactory2);
+   }
+
+//   public void testHotRodPutInNonOwner() {
```

Comment:
Are we adding this back in?  If not we can remove the SplitMagicKey class it seems.